src/openrct2/core/FixedPoint.hpp

```diff
@@ -21,7 +21,23 @@ using fixed32_2dp = int32_t;
 using fixed64_1dp = int64_t;
 
 // Construct a fixed point number. For example, to create the value 3.65 you
-// would write FIXED_2DP(3,65)
-#define FIXED_XDP(x, whole, fraction) ((whole) * (10 * (x)) + (fraction))
-#define FIXED_1DP(whole, fraction) FIXED_XDP(1, whole, fraction)
-#define FIXED_2DP(whole, fraction) FIXED_XDP(10, whole, fraction)
+// would write MakeFixed2dp(3, 65)
+template<typename T, uint8_t numDecimals>
+static constexpr T MakeFixedXdp(T whole, uint8_t fraction)
+{
+    static_assert(numDecimals > 0);
+    constexpr auto x = 10 ^ (numDecimals - 1);
```

Comment:
Why was this turned into a xor operation?

Reply:
... your question actually answers one I had. It is supposed to be "power of" rather than "xor".